### PR TITLE
IPAM tests

### PR DESCRIPTION
**Pull request by fcrisciani**
Added tests for swarm mode and also some new parallel tests with various subnets

Signed-off-by: Flavio Crisciani <flavio.crisciani@docker.com>

### CODE REVIEW

**Comment by fcrisciani** [comment]
Tests are expected to fail

**Comment by abhi** [comment]
@mghazizadeh @antonybichon17 @andrewhsu This is one of the test scenarios that reproduces the issue. The fix and additional tests are available at #2105.
Each of these test and the ones in #2105 should ideally fail consistently without the fix.

**Comment by fcrisciani** [comment]
ping @aboch

**Comment by fcrisciani** [comment]
To not lose the failure on master: https://circleci.com/gh/docker/libnetwork/5992

File: bitseq/sequence.go [hunk 1]
```diff
@@ -313,14 +319,14 @@ func (h *Handle) set(ordinal, start, end uint64, any bool, release bool, serial
 		curr := uint64(0)
 		h.Lock()
 		store = h.store
-		h.Unlock()
 		if store != nil {
+			h.Unlock() // The lock is acquired in the GetObject
```
Comment (ddebroy) [inline bitseq/sequence.go]: It appears `store.GetObject` will modify `h` (through the `SetValue` and `SetIndex` calls). If so, shouldn't we keep `h` locked while that is happening?

**Comment by abhi** [comment]
@ddebroy I have added that scenario implicitly in that test where it moves on to the next block because of the invalidPos. Are you suggesting we explicitly add it as well ?

**Comment by ddebroy** [comment]
Right ... that was my suggestion. Not a high priority and can be added later for completeness since it's implicitly covered right now as you mentioned.

File: vendor.conf [hunk 2]
```diff
@@ -50,5 +50,6 @@ github.com/vishvananda/netns 604eaf189ee867d8c147fafc28def2394e878d25
 golang.org/x/crypto 558b6879de74bc843225cde5686419267ff707ca
 golang.org/x/net 7dcfb8076726a3fdd9353b6b8a1f1b6be6811bd6
 golang.org/x/sys 07c182904dbd53199946ba614a412c61d3c548f5
+github.com/golang/sync fd80eb99c8f653c847d294a001bdf2a3a6f768f5
```
Comment (thaJeztah) [inline vendor.conf]: This uses the wrong (non-canonical) import path; see https://github.com/docker/libnetwork/pull/2111 (and https://github.com/moby/moby/pull/36589/files#r174434213)

**Pull request merged**